HW-25-CoreData/Source/Modules/EditScreen/View/EditViewController.swift

```diff
@@ -0,0 +1,223 @@
+        photo.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        photo.widthAnchor.constraint(equalToConstant: view.frame.size.width / 2).isActive = true
+        photo.heightAnchor.constraint(equalTo: photo.widthAnchor).isActive = true
+        
+        mainStackView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        mainStackView.topAnchor.constraint(equalTo: photo.bottomAnchor, constant: 40).isActive = true
+        mainStackView.widthAnchor.constraint(equalTo: view.widthAnchor, constant: -80).isActive = true
+        mainStackView.heightAnchor.constraint(equalToConstant: 210).isActive = true
+        
+        nameStack.translatesAutoresizingMaskIntoConstraints = false
+        nameStack.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        
+        ageStack.translatesAutoresizingMaskIntoConstraints = false
+        ageStack.heightAnchor.constraint(equalTo: nameStack.heightAnchor).isActive = true
+        
+        genderStack.translatesAutoresizingMaskIntoConstraints = false
+        genderStack.heightAnchor.constraint(equalTo: nameStack.heightAnchor).isActive = true
+    }
+    
+    //MARK: - ButtonAction functions
+    
+    @objc func editButtonAction() {
+        changeEditButton()
+    }
+    
+    @objc func backButtonAction() {
+        presenter?.returnMainView()
+    }
+    
+    private func editButtonPressed() {
+        editButton.title = isEditMode ? "Save" : "Edit"
+        mainStackView.isUserInteractionEnabled = true
+        print("Edit pressed")
+    }
+    
+    private func saveButtonPressed() {
+        if let isEmpty = nameStack.textField?.text?.isEmpty {
+            name = !isEmpty ? nameStack.textField?.text : nameStack.textField?.placeholder
+        }
+        age = ageStack.dataPicker?.date
+        mainStackView.isUserInteractionEnabled = false
+        print("save pressed")
+        updateInfo()
+    }
+    
+    private func changeEditButton() {
+        isEditMode.toggle()
+        isEditMode ? editButtonPressed() : saveButtonPressed()
+        editButton.title = isEditMode ? "Save" : "Edit"
+    }
+}
+
+//MARK: - Protocol methods
+
+extension EditViewController: EditViewProtocol {
+    func showModelInfo(_ friend: Friend) {
+        nameStack.textField?.placeholder = friend.name
+        ageStack.dataPicker?.date = friend.age ?? Date.init(timeIntervalSince1970: 1123123123)
+        switch friend.gender {
+        case "Мужчина":
+            genderStack.picker?.selectRow(0, inComponent: 0, animated: true)
+        case "Женьщина":
+            genderStack.picker?.selectRow(2, inComponent: 0, animated: true)
+        default:
+            genderStack.picker?.selectRow(1, inComponent: 0, animated: true)
+        }
+    }
+    
+    func updateInfo() {
+        guard let name = name else { return }
+        guard let age = age else { return }
+        guard let gender = gender else { return }
```

Comment:
Здесь можно через запятую перечислять
```suggestion
        guard let name = name, let age = age, let gender = gender else { return }
```